### PR TITLE
.NET9 rc2

### PR DESCRIPTION
**Pull request by alexis-kochetov**

### CODE REVIEW

**Review by frolyo — APPROVED** [review]

File: server-publish.cmd [hunk 1]
```diff
@@ -1,9 +1,9 @@
 :<<BATCH
     @echo off
-    dotnet publish src/dotnet/App.Server/App.Server.csproj -c:Release -- %*
+    dotnet publish src/dotnet/App.Server/App.Server.csproj -c:Release -p:UseAppPack=true -- %*
```
Comment (alexyakunin) [inline server-publish.cmd:3]: We should keep in mind this is one of changes here.

File: src/dotnet/App.Maui/App.Maui.csproj [hunk 2]
```diff
@@ -245,7 +241,7 @@
     <AndroidLinkTool>r8</AndroidLinkTool>
     <PublishTrimmed>true</PublishTrimmed>
     <TrimMode Condition="!$(UseAppPack)">partial</TrimMode>
-    <TrimMode Condition="$(UseAppPack)">partial</TrimMode> <!-- Will be "full" w/ .NET 9 -->
+    <TrimMode Condition="$(UseAppPack)">full</TrimMode>
```
Comment (alexyakunin) [inline src/dotnet/App.Maui/App.Maui.csproj]: And that's another one.

File: src/dotnet/App.Maui/App.Maui.csproj [hunk 3]
```diff
@@ -362,16 +358,16 @@
   </PropertyGroup>
 
   <!-- iOS: Debug configuration -->
-  <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net8.0-ios|AnyCPU'">
+  <PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net9.0-ios|AnyCPU'">
     <CreatePackage>false</CreatePackage>
-    <CodesignProvision>VS: chat.actual.dev.app Development</CodesignProvision>
+    <CodesignProvision>chat.actual.dev.app</CodesignProvision>
```
Comment (alexyakunin) [inline src/dotnet/App.Maui/App.Maui.csproj:363]: Intended?

File: src/dotnet/App.Maui/App.Maui.csproj [hunk 4]
```diff
@@ -381,8 +377,35 @@
     <UseInterpreter>true</UseInterpreter>
     <!--<MTouchInterpreter>-ActualChat,ActualChat.Chat.Contracts,ActualChat.Users.Contracts,ActualChat.Contacts.Contracts,ActualLab.Interception</MTouchInterpreter>-->
     <CreatePackage>false</CreatePackage>
+    <TrimmerSingleWarn>false</TrimmerSingleWarn>
```
Comment (alexyakunin) [inline src/dotnet/App.Maui/App.Maui.csproj]: Maybe makes sense to remove this line.

File: src/dotnet/App.Maui/App.Maui.csproj [hunk 5]
```diff
@@ -381,8 +377,35 @@
...
   </PropertyGroup>
 
+  <PropertyGroup Condition="$(TargetFramework.Contains('-windows')) and '$(Configuration)' == 'Release'">
+    <RuntimeIdentifier>win-x64</RuntimeIdentifier>
+    <PublishReadyToRun>true</PublishReadyToRun>
+<!--    <PublishSingleFile>true</PublishSingleFile>-->
+<!--    <WindowsAppSDKSelfContained>true</WindowsAppSDKSelfContained>-->
+    <SelfContained>true</SelfContained>
+    <PublishAot>false</PublishAot>
+    <PublishTrimmed>false</PublishTrimmed>
```
Comment (alexyakunin) [inline src/dotnet/App.Maui/App.Maui.csproj:395]: Can we trim windows app in partial mode?

Comment (alexyakunin) [reply src/dotnet/App.Maui/App.Maui.csproj:395]: Or doesn't make sense as well?

File: src/dotnet/App.Maui/App.Maui.csproj [hunk 6]
```diff
@@ -381,8 +377,35 @@
...
+    <TrimMode Condition="$(UseAppPack)">partial</TrimMode>
+    <OptimizationPreference>Size</OptimizationPreference>
+    <IlcGenerateStackTraceData>true</IlcGenerateStackTraceData>
+    <IlcDisableReflection>false</IlcDisableReflection>
+    <NativeDebugSymbols>false</NativeDebugSymbols>
+    <StripSymbols>false</StripSymbols>
+    <TrimmerSingleWarn>false</TrimmerSingleWarn>
+  </PropertyGroup>
+
+  <ItemGroup><IlcArg Include="--maxgenericcycle:1" /></ItemGroup>
```
Comment (alexyakunin) [inline src/dotnet/App.Maui/App.Maui.csproj]: Interesting whether this is necessary as well.

Comment (alexis-kochetov) [reply src/dotnet/App.Maui/App.Maui.csproj]: For diagnostics purposes - when was unable to build NativeAOT
https://github.com/dotnet/runtime/issues/97235

**Pull request merged**